commerce_coordinator/apps/commercetools/tasks.py

```diff
@@ -70,31 +63,20 @@ def _log_info_and_release_lock(log_message):
 
     try:
         client = CommercetoolsAPIClient()
-        current_order_version = order_version
+        order = client.get_order_by_id(order_id)
+        current_order_version = order.version
 
-        cache_entry = cache.get(cache_key, None)
-
-        # TODO: Remove logging after testing on stage
-        if cache_entry:
-            logger.info(f'[CT-{tag}] Found cache entry for order version {cache_entry} for cache key {cache_key}')
-            current_order_version = cache_entry
-        else:
-            logger.info(f'[CT-{tag}] Cache entry not found for order version for cache key {cache_key}')
-
-        if self.request.retries > 0:
-            current_order_version = client.get_order_by_id(order_id).version
+        line_item = get_edx_line_item(order.line_items, line_item_id)
 
         updated_order = client.update_line_item_on_fulfillment(
             entitlement_uuid,
             order_id,
             current_order_version,
             line_item_id,
-            item_quantity,
-            from_state_id,
+            line_item.quantity,
+            get_edx_line_item_state(line_item),
```

Comment:
We can remove this utility and assign the direct value here since this utility does not do anything specific.